File: chroma-league/src/test/java/net/booone/chromaleague/hud/animations/ActivePlayerAssistAnimationTest.java
package net.booone.chromaleague.hud.animations;

import net.booone.chromaleague.GameStateMocks;
import net.booone.chromaleague.hud.AnimationTester;
import org.junit.jupiter.api.Test;

class ActivePlayerAssistAnimationTest {

    @SuppressWarnings("JUnitTestMethodWithNoAssertions")
    @Test
    void playActivePlayerAssistAnimation() {
        //given
        new GameStateMocks().mockTestPlayerList();
        //then
        new AnimationTester().testAnimation(new ActivePlayerAssistAnimation());
    }
}

File: test/hud/animations/ActivePlayerAssistAnimation.test.ts
import {describe, test} from 'vitest';
import {AnimationTester} from '../../AnimationTester.js';
import {GameStateMocks} from '../../GameStateMocks.js';
import {ActivePlayerAssistAnimation} from '../../../src/hud/animations/ActivePlayerAssistAnimation.js';

/**
 * Integration test for active player assist animation on actual hardware.
 */
describe('ActivePlayerAssistAnimation Integration', () => {
    test('playActivePlayerAssistAnimation', async () => {
        //given
        new GameStateMocks().playerList();
        //then
        await new AnimationTester().testAnimation(new ActivePlayerAssistAnimation());
    }, 30000);
});
